bf2fc6cc711aee1a0c2a/kas-fleet-manager
# [MGDSTRM-2213] - Added terms acceptance middleware

## Description
Solve issue [MGDSTRM-2213](https://issues.redhat.com/browse/MGDSTRM-2213)

- a new middleware has been created to check that the user has accepted the terms. Terms are checked only for kafka creation
- a new configuration key has been added to the server configuration to indicate if we are OSS or product. The name of the key is `EnableTermsAcceptance`

## Verification Steps

1. Set `EnableTermsAcceptance` to `true`
2. Copy `ocm-service.clientId` and `ocm-service.clientSecret` from the [vault](https://vault.devshift.net/ui/vault/secrets/app-interface/show/managed-service-api/stage/service/credentials) into the corresponding file
3. Run the server
4. Try to call the `create kafka` endpoint and check that you get a **Forbidden** error if the terms must be accepted and your user has not accepted them yet.
**Example**:
```bash
    curl --request \
        POST -H "Authorization: Bearer $(ocm token)" \
        --data '{"region":"us-east-1","cloud_provider": "aws","name": "serviceapitest4", "multi_az":true}' \
        http://localhost:8000/api/managed-services-api/v1/kafkas\?async\=true
```

## Type of change


## Checklist (Definition of Done)
